modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -162,8 +163,8 @@ private <T> List<T> fetchDatabasesOverview() throws IOException {
     void processDatabase(Map<String, Object> databaseInfo) {
         String name = databaseInfo.get("name").toString().replace(".tgz", "") + ".mmdb";
         String md5 = (String) databaseInfo.get("md5_hash");
-        if (state.contains(name) && Objects.equals(md5, state.get(name).md5())) {
-            updateTimestamp(name, state.get(name));
+        if (state.get().contains(name) && Objects.equals(md5, state.get().get(name).md5())) {
+            updateTimestamp(name, state.get().get(name));
```

Comment:
Is this method guarded by a lock? If not then above 3 `state.get()` might produce different results as well?